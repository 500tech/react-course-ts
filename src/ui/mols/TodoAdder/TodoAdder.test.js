import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TodoAdder } from '.';

test('Nothing is broken', () => {
  const wrapper = mount(<TodoAdder />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

test('Initializing text initializes text', () => {
  const wrapper = mount(<TodoAdder initialText="initial text" />);
  expect(wrapper.find('input').props().value).toBe('initial text');
  expect(wrapper.instance().state.text).toBe('initial text');
});

test('Autofocus', () => {
  const wrapper = mount(<TodoAdder initialText="initial text" />);
  expect(wrapper.instance().inputRef.current).not.toBeNull();
  expect(wrapper.instance().inputRef.current).toBe(
    window.document.activeElement
  );
});

test('Submit', () => {
  const onAdd = jest.fn();
  const wrapper = mount(<TodoAdder initialText="initial text" onAdd={onAdd} />);
  wrapper.find('form').simulate('submit');
  expect(wrapper.instance().state.text).toBe('');
  expect(onAdd).toHaveBeenCalledWith('initial text');
});
