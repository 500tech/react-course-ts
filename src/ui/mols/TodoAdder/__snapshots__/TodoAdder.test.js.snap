// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nothing is broken 1`] = `
<TodoAdder>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value=""
    />
    <button
      disabled={true}
    >
      Add
    </button>
  </form>
</TodoAdder>
`;
